Does not compile with GCC 11

When trying to compile with GCC 11, I get the following error:

```
../../../crazyflies/crazyflie-firmware/src/deck/core/deck_test.c: In function 'decktestSaveGPIOStatesABC':
../../../crazyflies/crazyflie-firmware/src/deck/core/deck_test.c:61:3: error: 'memcpy' offset [0, 39] is out of the bounds [0, 0] [-Werror=array-bounds]
   61 |   memcpy(&gpioRegBuf->gpioBuffA, GPIOA, sizeof(GPIO_TypeDef));
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../crazyflies/crazyflie-firmware/src/deck/core/deck_test.c:62:3: error: 'memcpy' offset [0, 39] is out of the bounds [0, 0] [-Werror=array-bounds]
   62 |   memcpy(&gpioRegBuf->gpioBuffB, GPIOB, sizeof(GPIO_TypeDef));
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../crazyflies/crazyflie-firmware/src/deck/core/deck_test.c:63:3: error: 'memcpy' offset [0, 39] is out of the bounds [0, 0] [-Werror=array-bounds]
   63 |   memcpy(&gpioRegBuf->gpioBuffC, GPIOC, sizeof(GPIO_TypeDef));
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../crazyflies/crazyflie-firmware/src/deck/core/deck_test.c: In function 'decktestRestoreGPIOStatesABC':
../../../crazyflies/crazyflie-firmware/src/deck/core/deck_test.c:69:3: error: 'memcpy' offset [0, 39] is out of the bounds [0, 0] [-Werror=array-bounds]
   69 |   memcpy(GPIOA, &gpioRegBuf->gpioBuffA, sizeof(GPIO_TypeDef));
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../crazyflies/crazyflie-firmware/src/deck/core/deck_test.c:70:3: error: 'memcpy' offset [0, 39] is out of the bounds [0, 0] [-Werror=array-bounds]
   70 |   memcpy(GPIOB, &gpioRegBuf->gpioBuffB, sizeof(GPIO_TypeDef));
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../crazyflies/crazyflie-firmware/src/deck/core/deck_test.c:71:3: error: 'memcpy' offset [0, 39] is out of the bounds [0, 0] [-Werror=array-bounds]
   71 |   memcpy(GPIOC, &gpioRegBuf->gpioBuffC, sizeof(GPIO_TypeDef));
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[1]: *** [../../../crazyflies/crazyflie-firmware/tools/make/targets.mk:27: deck_test.o] Error 1
make: *** [../../../crazyflies/crazyflie-firmware/Makefile:400: build] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -316,6 +316,12 @@ endif
 # Disable warnings for unaligned addresses in packed structs (added in GCC 9)
 CFLAGS += -Wno-address-of-packed-member
 
+# Disable warnings for incorrectly detected region size (added in GCC 11)
+# The compiler is not detecting properly GPIO structure size
+CFLAGS += -Wno-array-bounds
+CFLAGS += -Wno-stringop-overread
+CFLAGS += -Wno-stringop-overflow
+
 ifeq ($(LTO), 1)
   CFLAGS += -flto
 endif

